src/session_pool/session_pool.js

```diff
@@ -172,12 +172,17 @@ export class SessionPool extends EventEmitter {
      * If there is space for new session, it creates and return new session.
      * If the session pool is full, it picks a session from the pool,
      * If the picked session is usable it is returned, otherwise it creates and returns a new one.
-     *
+     * @param {String} sessionId
      * @return {Promise<Session>}
      */
-    async getSession() {
+    async getSession(sessionId) {
+        if (sessionId) {
+            const foundSession = this.sessions.find(session => session.id === sessionId);
+            if (foundSession) return foundSession;
+        }
```

Comment:
This is not very performant. If we decide to do this, we should change the way `sessions` are stored in `SessionPool` so that this is an O(1) operation and not O(n).

Reply:
Yep, not great with many sessions. Maybe a tree or hash table? I'm not very knowledgeable in these.